File: main/src/main/java/net/fwitz/math/main/binary/complex/functions/examples/simple/MultiplePole.java
package net.fwitz.math.main.binary.complex.functions.examples.simple;

import net.fwitz.math.plot.binary.complex.color.DomainColoringAdvancedPlusLogScale;
import net.fwitz.math.plot.binary.complex.ComplexFunctionPlot;

public class MultiplePole {
    public static void main(String[] args) {
        new ComplexFunctionPlot("z ^ -3")
                .computeFn(z -> z.pow(-3.0))
                .colorFn(new DomainColoringAdvancedPlusLogScale())
                .domainX(-1.5, 1.5)
                .domainY(-1.5, 1.5)
                .render();
    }
}

File: main/src/main/kotlin/net/fwitz/math/main/binary/complex/functions/examples/simple/MultiplePole.kt
package net.fwitz.math.main.binary.complex.functions.examples.simple

import net.fwitz.math.plot.binary.complex.ComplexFunctionPlot
import net.fwitz.math.plot.binary.complex.color.DomainColoringAdvancedPlusLogScale

object MultiplePole {
    @JvmStatic
    fun main(args: Array<String>) = ComplexFunctionPlot("z ^ -3")
        .computeFn { z -> z.pow(-3.0) }
        .colorFn(DomainColoringAdvancedPlusLogScale)
        .domainX(-1.5, 1.5)
        .domainY(-1.5, 1.5)
        .render()
}